policies/verify-tenant-patch/policies/rule_rego_test.rego

```diff
@@ -0,0 +1,650 @@
+			"metadata": {"creationTimestamp": "2020-12-25T16:16:14Z", "generation": 1, "name": "test-name", "uid": "c3b3a51a-1a43-47f2-a7f4-c84914b026ba"},
+			"spec": {
+				"email": "test.name@email.com", "firstName": "test-name", "lastName": "test-last", "publicKeys": ["key1", "key2", "key3"],
+				"workspaces": [{"groupNumber": 12, "role": "manager", "workspaceRef": {"name": "test-land"}}],
+			},
+		},
+		"oldObject": null,
+		"userInfo": {"username": "mariocalza", "groups": ["system:masters", "system:authenticated"]},
+	}
+}
+
+input_review_invalid_creation_creator_not_manager = output {
+	output = {
+		"requestKind": {"group": "crownlabs.polito.it", "version": "v1alpha1", "kind": "Tenant"}, "name": "test-name", "operation": "CREATE",
+		"object": {
+			"apiVersion": "crownlabs.polito.it/v1alpha1", "kind": "Tenant",
+			"metadata": {"creationTimestamp": "2020-12-25T16:16:14Z", "generation": 1, "name": "test-name", "uid": "c3b3a51a-1a43-47f2-a7f4-c84914b026ba"},
+			"spec": {
+				"email": "test.name@email.com", "firstName": "test-name", "lastName": "test-last", "publicKeys": ["key1", "key2", "key3"],
+				"workspaces": [{"groupNumber": 12, "role": "manager", "workspaceRef": {"name": "landc"}}],
+			},
+		},
+		"oldObject": null,
+		"userInfo": {"username": "mariocalza", "groups": ["system:masters", "system:authenticated"]},
+	}
+}
+
+input_review_valid_creation_tenant = output {
+	output = {
+		"requestKind": {"group": "crownlabs.polito.it", "version": "v1alpha1", "kind": "Tenant"}, "name": "test-name", "operation": "CREATE",
+		"object": {
+			"apiVersion": "crownlabs.polito.it/v1alpha1", "kind": "Tenant",
+			"metadata": {"creationTimestamp": "2020-12-25T16:16:14Z", "generation": 1, "name": "test-name", "uid": "c3b3a51a-1a43-47f2-a7f4-c84914b026ba"},
+			"spec": {
+				"email": "test.name@email.com", "firstName": "test-name", "lastName": "test-last", "publicKeys": ["key1", "key2", "key3"],
+				"workspaces": [{"groupNumber": 12, "role": "user", "workspaceRef": {"name": "landc"}}],
+			},
+		},
+		"oldObject": null,
+		"userInfo": {"username": "mariorossi", "groups": ["system:masters", "system:authenticated"]},
+	}
+}
+
+
+input_review_valid_creation_cluster_admin=output{
+        output = {
+		"requestKind": {"group": "crownlabs.polito.it", "version": "v1alpha1", "kind": "Tenant"}, "name": "test-name", "operation": "CREATE",
+		"object": {
+			"apiVersion": "crownlabs.polito.it/v1alpha1", "kind": "Tenant",
+			"metadata": {"creationTimestamp": "2020-12-25T16:16:14Z", "generation": 1, "name": "test-name", "uid": "c3b3a51a-1a43-47f2-a7f4-c84914b026ba"},
+			"spec": {
+				"email": "test.name@email.com", "firstName": "test-name", "lastName": "test-last", "publicKeys": ["key1", "key2", "key3"],
+				"workspaces": [{"groupNumber": 12, "role": "user", "workspaceRef": {"name": "landa"}}],
+			},
+		},
+		"oldObject": null,
+		"userInfo": {"username": "mariorossi", "groups": ["system:masters", "system:authenticated","kubernetes:admin"]},
```

Comment:
I would use a different username or workspace in this test, as IIUC, this user can already manage this course according to his Tenant resource.